### ISSUE TITLE
blockHeader function improvement

### ISSUE THREAD

**Issue by doiftrue** [erusev/parsedown#362]
Look to this example, assume we trying to parse this header:
`### #header text`
We get `<h3>header text</h3>` but not `<h3>#header text</h3>` as it's on github or anywhere else. It's because of this line in `blockHeader` function:

```
$text = trim($Line['text'], '# ');
```

I suggest replace it with two lines:

```
$text = trim($Line['text'], '#');
$text = trim($text);
```

Thanks for code and hope you repair this in next release.

**Comment by kminchev** [comment]
Workaround is `\#header`

**Comment by doiftrue** [comment]
Thanks, I know about `\#header` but it's bad option for me. I make wrapper class for parsedown and add my blockHeader method.

For example here on github if a write `### #header` i get `#header` not `header` and I think it's more correctly.

**Comment by erusev** [comment]
@doiftrue it is indeed more correct - i'll fix it as soon as i get a chance

**Comment by doiftrue** [comment]
Thanks a lot!

**Comment by praveenscience** [comment]
Guess in simple terms, we can make it using double `trim()` like this:

    $text = trim(trim($Line['text'], '#'));

**Comment by aidantwoods** [comment]
Looks like I added an equivalent fix in #583, this will be in the next version :)
